### PR TITLE
Sortable table performance fixes

### PR DESCRIPTION
**Pull request by nwmac**
Note: This PR should be merged after https://github.com/rancher/dashboard/pull/5382

This PR improves the performance of Sortable Tables.

- Live Columns (see detail below)
- Delayed Columns (see detail below)
- Reduce default page size to 50
- Created a new computable `displayRows` which efficiently collects all of the data for the table display ensuring that we do not call the expensive `valueFor` method more then once per column
- Improves efficiency of the `valueFor` method (see below)
- Adds a mechanism where a column can be hidden for large page sizes
- Hides the endpoints column on the workload table when the page size is large (GitHub issue created to show this information on the detail view for workloads)
- Improved the efficiency of custom component creation for columns by not using component name but using component reference (see FORMATTERS in the sortable table index.vue)

**Live Columns**
This PR changes the way that the LiveDate column works (and provides a mechanism we can use for all Live columns).

Currently, when there are 100 rows in a page on a sortable table with the Age column, we create a timer for every age field.

On some older browsers (Edge 91, for example), this makes it impossible to use the scroll bar to scroll through the list.

This PR makes 5 changes to support live columns:

1. Provides a mechanism where the SortableTable will use a single timer to update all live columns in the table
2. Updates the LiveDate component to use this mechanism
3. Updates SortableTable so that we pause live updating when the user is scrolling the page
4. Slightly improves the performance of the LiveDate component
5. Only updates the visible columns rather than the whole list

**Delay-loaded Columns**
In addition to live columns, this PR allows a column to be delay-loaded and for a custom formatter component to support delay loading.

The mechanism is similar to live columns:

- There will be a delay before these columns are rendered to give the table time to render
- Components are updated in batches
- Only visible columns are loaded
- Renders newly-visible columns on scroll and resize
- Updated the WorkloadHealthScale component to be delay-loaded
- Created basic developer doc

**valueFor**

The `valueFor` method which gets the value for a column is expensive.

If takes an object and a string and has two main ways of getting the value.

1. When the string is of the form `field.field.field` it uses `get` from object utils, which in turn splits the string and evaluates this
2. When the string starts with a `$` character it uses JSONPath to evaluate the value - this is slow

This PR:

- Ensures we only call `valueFor` is little as possible
- Adds support for `value` on a column definition being a function, allowing columns to provide the function to get a value for a row - this is the most efficient mechanism. These will be stored in the store, but I don't think that is an issue.
- Updates some columns definitions for slow tables to use value getters for columns
- When a column definition is just a string, e.g. `Ready`, the code uses the schema definition and will use a value reference of the form `$.`, which is expensive - this is now detected and we use a generated function to get the indexed field

### CODE REVIEW

File: components/SortableTable/index.vue [hunk 1]
```diff
@@ -662,48 +838,73 @@ export default {
...
-                      :width="col.width"
+                      :key="col.col.name"
+                      :data-title="col.col.label"
+                      :align="col.col.align || 'left'"
+                      :class="{['col-'+col.dasherize]: !!col.col.formatter, [col.col.breakpoint]: !!col.col.breakpoint, ['skip-select']: col.col.skipSelect}"
+                      :width="col.col.width"
                     >
-                      <slot :name="'cell:' + col.name" :row="row" :col="col" :value="valueFor(row,col)">
+                      <slot :name="'cell:' + col.name" :row="row" :col="col" :value="col.value">
+                        <span v-if="col.formatter && col.formatter === 'LinkDetail'">
```
Comment (richard-cox) [inline components/SortableTable/index.vue:870]: Should this be removed? In theory it should be col.col.formatter... however that breaks as most don't have a `url`, it ignores the `to()` calc from `LinkDetail` and the `value` is a col.col.value which is a function. 
I was testing  with the Machine Pools page for an imported / rke1 cluster

File: components/SortableTable/index.vue [hunk 2]
```diff
@@ -721,17 +922,17 @@ export default {
             </slot>
           </slot>
           <slot
-            v-if="showSubRow(row, keyField)"
+            v-if="row.showSubRow"
```
Comment (richard-cox) [inline components/SortableTable/index.vue:925]: Note to self - I was using the Nodes list with an added taint to test

**Review by richard-cox — CHANGES_REQUESTED** [review]
- Filtering does not work. Enter filter `a` in Workloads view causes list to flash. filter is added to query param. filter is removed from input field. also on refresh query param is not shown in input field
- ~~In the workloads view, with 62 jobs, 14 deployments and a few other types only 20 resources are shown (there's no project/namespace filter or text filter). Setting the page count to 10 shows pagination text of `1 - 10 of 20 Workloads`~~ user error!
- storing functions in the store means only one thing...
- this is a real stretch goal, but could we load the next N (1-5) workload health components such that slowly scrolling down the list doesn't result in flashing of the workload health loading indicators?
- In workloads when grouped by namespace the group label is shown fine. In the project/namespace list the group label is missing
- I tested the reactivity of lists a lot (rows/subrows updated, rows added/removed, etc) everything looks good there

File: components/SortableTable/index.vue (hunk 1 above)
Comment (nwmac) [reply components/SortableTable/index.vue:870]: Yes it should - this was testing inline a formatter - not needed

**Comment by nwmac** [comment]
@richard-cox PRs comments should be addressed now

**Review by richard-cox — CHANGES_REQUESTED** [review]
When a page is loaded that contains a `page=<x>` query param the list doesn't contain any rows 
- In master the `page` & `sort` are totally ignored and removed on load
- In this branch and master the `sort` and order direction `desc/asc` params are honoured 

Going on discussions we should remove page and sort query params. They're not working in master, cause performance issues if working and break multi-list pages

Other than that looks good

**Review by richard-cox — APPROVED** [review]
There's a couple of tech debt things, but this can merge without those tweaks
- There's some references to SEARCH_QUERY in the app chart list. That side still works though (query string applied on refresh)
- the query param consts can be removed from `config/query-params.js`

**Pull request merged**